mailbox/api/src/main/java/org/apache/james/mailbox/model/MessageRange.java

```diff
@@ -129,6 +130,12 @@ public long getUidTo() {
         return uidTo;
     }
 
+    public long size() {
+        if (type == Type.FROM || type == Type.ALL) {
```

Comment:
Nope. Type FROM indicates that this MessageRange contains all uids of the mailbox superior to from date.
Thus, we can not know the size of this message range.
By the way, after reading matthieu comments, I realized they were no need to hndle deletes using MessageRange if I reworl ListeningSearchIndex class a little. I deleted this changeset and will propose a new one, of higher quality.